src/preview.cpp

```diff
@@ -803,13 +804,14 @@ namespace {
                 else {
                     // FIXME: the buffer is probably copied twice, it should be optimized
                     DataBuf buf(size_);
-                    Exiv2::byte* pos = buf.pData_;
+                    uint32_t idxBuf = 0;
                     for (int i = 0; i < sizes.count(); i++) {
                         uint32_t offset = dataValue.toLong(i);
                         uint32_t size = sizes.toLong(i);
-                        if (offset + size <= static_cast<uint32_t>(io.size()))
-                            memcpy(pos, base + offset, size);
-                        pos += size;
+                        enforce(idxBuf + size < size_, kerCorruptedMetadata);
+                        if (size!=0 && offset + size <= static_cast<uint32_t>(io.size()))
+                            memcpy(reinterpret_cast<byte *>(buf.pData_[idxBuf]), base + offset, size);
```

Comment:
Uhm, are you sure about this one? `buf.pData_[idxBuf]` is a `byte`, not `byte *`. Wont this wreak havoc as this code now tries to copy into the memory address stored in `buf.pData_[idxBuf]`?

Reply:
of fuck, what the hell I did there ... you are totally right, I need to simply use there `&(buf.pData_[idxBuf]`